коммит рабочего приложения

sds96 opened the pull request:
Есть пара вопросов, примечаний:

1) Очень странное решение хранить список покупок в строке - как рекомендуют авторы. Да, структура такова, что массивы "мы ещё не знаем", но тогда и смысла в отдельном классе для покупок ровно нуль.
2) Есть ли пути обработки неправильного ввода помимо исключений?

sds96 committed:
коммит рабочего приложения

Charmandik commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,74 @@
+        System.out.println("Давайте добавим ваши товары.");
+        while(true){
+            System.out.println("Введите название товара или 'Завершить' если вы закончили.");
+            String name = scanner.next();
```
В текущей реализации, при вводе названия товара, программа использует scanner.next(), что ограничивает ввод до первого пробела. Если название товара состоит из нескольких слов, только первое слово будет использовано.

Charmandik commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,74 @@
+            }
+
+            totalSum += value;
+            allPurchasees = allPurchasees.concat(String.format("\n%s %.2f", name, value));
```
При использовании concat() в цикле добавления товаров, каждый раз создается новая строка, что может привести к неэффективному использованию памяти. Лучше использовать StringBuilder

Charmandik commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,74 @@
+            double value;
+            while (true){
+                System.out.println("Введите стоимость " + name + ":");
+                value = scanner.nextDouble();
```
После использования сканнера его следует закрыть с помощью scanner.close().

Charmandik commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,74 @@
+        System.out.println(String.format("Каждый гость должен заплатить %.2f %s", pricePerPerson, getPadezh(pricePerPerson)) );
+    }
+
+    String getPadezh(double d){
```
лучше всё же не использовать русские слова написанные английскими буквами, это может в превратиться в poluchitPadezh()  :D

Charmandik commented:
2. Можно использовать методы scanner.hasNextInt()/hasNextFloat()

sds96 committed:
Исправление ошибок.
Имя товара теперь может состоять из нескольких слов.
Правильный падеж для рублей 11-19.

sds96 committed:
StringBuilder для списка покупок

sds96 commented:
Поправил замечания
